src/week-view.c

```diff
@@ -480,17 +484,20 @@ void update_view_span(WeekView* wv)
 	wv->current_view = icaltime_span_new(start, until, 0);
 }
 
-GtkWidget* week_view_new()
+GtkWidget* week_view_new(icaltimetype today, int todays_week, int todays_year)
 {
 	WeekView* cw = g_object_new(FOCAL_TYPE_WEEK_VIEW, NULL);
 
 	char* zoneinfo_link = realpath("/etc/localtime", NULL);
 	cw->current_tz = icaltimezone_get_builtin_timezone(zoneinfo_link + strlen("/usr/share/zoneinfo/"));
 	free(zoneinfo_link);
 
-	icaltimetype today = icaltime_today();
-	cw->current_week = icaltime_week_number(today) + 1;
-	cw->current_year = today.year;
+	cw->todays_week = todays_week;
+	cw->todays_year = todays_year;
+
+	// initially show current week of current year
+	cw->current_week = todays_week;
+	cw->current_year = todays_year;
```

Comment:
Instead of setting these once at construction time, I suggest extending the `update_current_time` function to store more than the hour and the week day. This can then be used by the `week_view_current` function